Lib/traceback.py

```diff
@@ -476,34 +476,50 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
         _seen.add(id(exc_value))
         # Gracefully handle (the way Python 2.4 and earlier did) the case of
         # being called with no type or value (None, None, None).
+        self.__suppress_context__ = False
+        self._truncated = False
         if (exc_value and exc_value.__cause__ is not None
             and id(exc_value.__cause__) not in _seen):
```

Comment:
If recursion into constructing cause exceeded the recursion limit then surely the call to construct context will too. Why do we need to try blocks?